packages/shared/src/hooks/chat/useChatStream.ts

```diff
@@ -45,9 +45,10 @@ export const useChatStream = (): UseChatStream => {
       let queryKey: ReturnType<typeof generateQueryKey> = generateQueryKey(
         RequestKey.Search,
         user,
-        Date.now().toString(),
+        null,
```

Comment:
The last parameter is optional, right?

Reply:
So if we don't set anything then the query key would be `['search', userId]`, and when clearing it up with `removeQueries` it would remove all other queries related to search. That is why I added `null` to have I guess shared state for those initial errors (rate limit is the only one currently).

I see the bad side in this where if we used two streams in the same time they would share this error state. I do think its something we can sort when adding support for multiple streams in the same time in the future.

cc: @rebelchris since you also noted it.

Reply:
Hmm let's leave it as is for now then :)

Reply:
Merging this for now so it can be tested, if any further comments I also added a task https://dailydotdev.atlassian.net/browse/WT-1777

Reply:
Agreed let's see how it goes. We can probably use the session id for the last parameter to make it linked to that id only.

Reply:
We are using it, but on initial errors like rate limit there is no session id. That the only case where it is null for now.